bochs/bx_debug/dbg_main.cc

```diff
@@ -702,6 +702,9 @@ void bx_dbg_lin_memory_access(unsigned cpu, bx_address lin, bx_phy_address phy,
 
 void bx_dbg_phy_memory_access(unsigned cpu, bx_phy_address phy, unsigned len, unsigned memtype, unsigned rw, unsigned access, Bit8u *data)
```

Comment:
the 

```
  if (! bx_dbg.debugger_active)
    return;
```

is here. This means function call is always done (will all the parameters) and it costs a lot.
Can be optimized significantly by extracting the 'if' into BX_DBG_LIN_MEMORY_ACCESS and BX_DBG_PHY_MEMORY_ACCESS macros